aiida/tools/groups/grouppaths.py

```diff
@@ -0,0 +1,321 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Provides functionality for managing large numbers of AiiDA Groups, via label delimitation."""
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+from collections import namedtuple
+from functools import total_ordering
+import re
+from typing import Any, Iterable, List, Optional  # pylint: disable=unused-import
+import warnings
+
+from aiida import orm
+from aiida.common.exceptions import NotExistent
+
+__all__ = ('GroupPath', 'InvalidPath')
+
+REGEX_ATTR = re.compile('^[a-zA-Z][\\_a-zA-Z0-9]*$')
+
+
+class InvalidPath(Exception):
+    pass
+
+
+WalkNodeResult = namedtuple('WalkNodeResult', ['group_path', 'node'])
+
+
+@total_ordering
```

Comment:
It's purely a path string comparison, so that you can use `sorted(group_path.children)` and guarantee the order. As I've mentioned below, it should work generally the same as `pathlib.Path`, which in this case it does:

```python

In [0]: from pathlib import Path

In [1]: (Path('')/'a/f' ) > (Path('')/'a')
Out[1]: True

In [2]: (Path('')/'a/f' ) > (Path('')/'a')
Out[2]: True

In [3]: (Path('')/'a' ) < (Path('')/'a/c/e')
Out[3]: True

In [4]: (Path('')/'a/f' ) > (Path('')/'a/c/e')
Out[4]: True
```

Reply:
Ahh, I see, so it is just for the purpose of listing it in a user / performance friendly way. Ok then, forget I mentioned it.